tfhe/src/high_level_api/integers/types/compact.rs

```diff
@@ -90,3 +92,273 @@ where
+        };
+        self.list.is_conformant(&params)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::conformance::ParameterSetConformant;
+    use crate::core_crypto::prelude::UnsignedInteger;
+    use crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    use crate::shortint::{CiphertextModulus, PBSOrder};
+    use crate::{
+        generate_keys, set_server_key, CompactFheUint8, CompactFheUint8List, ConfigBuilder,
+    };
+    use rand::{thread_rng, Rng};
+
+    type ParameterAccessor<Ct, T> = dyn Fn(&mut Ct) -> &mut T;
+
+    type ParameterModifier<'a, Ct> = dyn Fn(&mut Ct) + 'a;
+
+    fn change_parameters<Ct, T: UnsignedInteger>(
+        func: &ParameterAccessor<Ct, T>,
+    ) -> Vec<Box<ParameterModifier<'_, Ct>>> {
+        vec![
+            Box::new(|ct| *func(ct) = T::ZERO),
+            Box::new(|ct| *func(ct) = func(ct).wrapping_add(T::ONE)),
+            Box::new(|ct| *func(ct) = func(ct).wrapping_sub(T::ONE)),
+        ]
+    }
+
+    #[test]
+    fn test_invalid_generic_compact_integer() {
+        type Ct = CompactFheUint8;
+
+        let config = ConfigBuilder::all_disabled()
+            .enable_default_integers()
+            .build();
+
+        let (client_key, _server_key) = generate_keys(config);
+
+        let public_key = CompactPublicKey::new(&client_key);
+
+        let ct = CompactFheUint8::try_encrypt(0, &public_key).unwrap();
+
+        assert!(ct.is_conformant(&PARAM_MESSAGE_2_CARRY_2_KS_PBS.into()));
+
+        let breaker_lists = vec![
+            change_parameters(&|ct: &mut Ct| &mut ct.list.num_blocks),
+            change_parameters(&|ct: &mut Ct| &mut ct.list.ct_list.message_modulus.0),
+            change_parameters(&|ct: &mut Ct| &mut ct.list.ct_list.carry_modulus.0),
+            change_parameters(&|ct: &mut Ct| &mut ct.list.ct_list.degree.0),
+            change_parameters(&|ct: &mut Ct| {
+                &mut ct.list.ct_list.ct_list.get_mut_lwe_ciphertext_count().0
+            }),
+            change_parameters(&|ct: &mut Ct| &mut ct.list.ct_list.ct_list.get_mut_lwe_size().0),
+        ];
+
+        for breaker_list in breaker_lists {
+            for breaker in breaker_list {
+                let mut ct_clone = ct.clone();
+
+                breaker(&mut ct_clone);
+
+                assert!(!ct_clone.is_conformant(&PARAM_MESSAGE_2_CARRY_2_KS_PBS.into()));
+            }
+        }
+
+        let breakers2: Vec<&ParameterModifier<'_, Ct>> = vec![
+            &|ct: &mut Ct| ct.list.ct_list.pbs_order = PBSOrder::BootstrapKeyswitch,
+            &|ct: &mut Ct| {
+                *ct.list.ct_list.ct_list.get_mut_ciphertext_modulus() =
+                    CiphertextModulus::try_new_power_of_2(1).unwrap();
+            },
+            &|ct: &mut Ct| {
+                ct.list.ct_list.ct_list.get_mut_container().pop();
+            },
+            &|ct: &mut Ct| ct.list.ct_list.ct_list.get_mut_container().push(0),
+        ];
+
+        for breaker in breakers2 {
```

Comment:
we are testing each breaker separately, is it possible to use an iproduct! on those breakers maybe ? i.e. some more complicated combination ?

otherwise we can go ahead with a first implementation and then strengthen testing with the composition of several breakers, it may become complicated to break the ciphertext and be sure it's not randomly valid, so yeah maybe not right away

Reply:
I agree on the principle

But it seems complicated and I don't really see the benefit it would provide

Reply:
I would say a follow up issue would be the way to go there

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/271